app/models/miq_request.rb

```diff
@@ -482,12 +482,12 @@ def self.update_request(request, values, requester)
   end
 
   def update_request(values, requester)
-    update_attribute(:options, options.merge(values))
-    set_description(true)
-
-    log_request_success(requester, :updated)
-
-    call_automate_event_queue("request_updated")
+    update_attributes(:options => options.merge(values))
+    if values[:user_message].present? && values.keys.length == 1
+      self.user_message = values[:user_message]
```

Comment:
You always want to call `user_message=` if values contains the `:user_message` key.  This logic only calls it if it is the only key.  Also, `user_message=` should be called with an empty string if that is passed.

I'm thinking it should be something like this:
```ruby
def update_request(values, requester)
  update_attributes(:options => options.merge(values))
  self.user_message = values[:user_message] unless values[:user_message].nil?
  after_update_options(requester) unless values.keys == [:user_message]

  self
end
```